timesketch/api/v1/resources/sigma.py

```diff
@@ -0,0 +1,220 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sigma resources for version 1 of the Timesketch API."""
+
+from flask import abort
+from flask_restful import Resource
+from flask_restful import reqparse
+from flask_login import login_required
+from flask_login import current_user
+from flask import jsonify
+from flask import current_app
+
+import six
+import os
+import codecs
+import yaml
+
+from timesketch.api.v1 import resources
+from timesketch.lib.definitions import HTTP_STATUS_CODE_OK
+from timesketch.lib.definitions import HTTP_STATUS_CODE_CREATED
+from timesketch.lib.definitions import HTTP_STATUS_CODE_BAD_REQUEST
+from timesketch.lib.definitions import HTTP_STATUS_CODE_FORBIDDEN
+from timesketch.lib.definitions import HTTP_STATUS_CODE_NOT_FOUND
+
+from sigma.backends import elasticsearch as sigma_elasticsearch
+import sigma.configuration as sigma_configuration
+from sigma.parser import collection as sigma_collection
+
+
+# TODO maybe do that moving forward and move code from here to the model
+#from timesketch.models.sigma import Sigma
+from timesketch.lib.analyzers import interface
+
+import logging
+
+logger = logging.getLogger('timesketch.api.sigma')
+
+class SigmaListResource(resources.ResourceMixin, Resource):
+    """Resource to get list of users."""
+
+    @login_required
+    def get(self):
+        """Handles GET request to the resource.
+
+        Returns:
+            Dict of sigma rules
+        """
+        sigma_rules = []
+        
+        # TODO: that should be part of the Timesketch config.
+        try:
+            _CONFIG_FILE = current_app.config['SIGMA_CONFIG']
+        except KeyError:
+            logger.error("SIGMA_CONFIG not found in config file")
+            _CONFIG_FILE = '../../../../data/sigma_config.yaml'
+
+        # Path to the directory containing the Sigma Rules to run, relative to
+        # this file.
+        # TODO check if get is better for getting the key errors
+        try:
+            _RULES_PATH = current_app.config['SIGMA_RULES_FOLDER']
+        except KeyError:
+            logger.error("SIGMA_RULES_FOLDER not found in config file")
+            _RULES_PATH = '../../../../data/sigma/rules/'
+
+        sigma_config_path = os.path.join(os.path.dirname(__file__), _CONFIG_FILE)
+
+        with open(sigma_config_path, 'r') as sigma_config_file:
```

Comment:
do we not want other types of verification that this is in fact a sigma config file?

you also just read in the entire file here? what if it is exceedingly large? no way to read that in chunks?

Reply:
hm yes I see what you are saying here, I have not seen a very very large Sigma config, and wrt "verification if is in fact a sigma config file", we could certainly check if it is a valid yml file and or what the beginnig of the file looks like, however I would kind of expect that from the sigma backend to do and "complain" if it is not a legit config file.

WDYT?